diff --git a/src/net/java/sip/communicator/impl/osdependent/systemtray/SystemTray.java b/src/net/java/sip/communicator/impl/osdependent/systemtray/SystemTray.java
index 125a190..6c5ab9b 100644
--- a/src/net/java/sip/communicator/impl/osdependent/systemtray/SystemTray.java
+++ b/src/net/java/sip/communicator/impl/osdependent/systemtray/SystemTray.java
@@ -19,7 +19,10 @@
 
 import javax.swing.*;
 
+import org.jitsi.util.OSUtils;
+
 import net.java.sip.communicator.impl.osdependent.systemtray.awt.*;
+import net.java.sip.communicator.impl.osdependent.systemtray.swingframe.SwingSystemTray;
 
 /**
  * Base class for all wrappers of <tt>SystemTray</tt> implementations.
@@ -38,7 +41,18 @@ public final static SystemTray getSystemTray()
         {
             if (java.awt.SystemTray.isSupported())
             {
-                systemTray = new AWTSystemTray();
+                if (OSUtils.IS_LINUX)
+                { // There are problems with AWTSystemTray on Linux.
+                    systemTray = new SwingSystemTray();
+                }
+                else
+                {
+                    systemTray = new AWTSystemTray();
+                }
+            }
+            else
+            {
+                systemTray = new SwingSystemTray();
             }
         }
 
